src/kube_state.rs

```diff
@@ -268,12 +268,19 @@ impl Resources {
         }
     }
 
-    fn upsert_node(&mut self, node: Node) -> Machine {
+    fn upsert_node(&mut self, node: Node) -> Option<Machine> {
```

Comment:
This method name threw me off. It's no longer idempotent if it's returning an Option, plus I tend to think of a database when I hear "upsert".